src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.cpp

```diff
@@ -63,7 +66,16 @@ std::tuple<int, evolution::dg::subcell::RdmpTciData> TciOnFdGrid::apply(
       subcell_mesh.extents(),
       evolution::dg::subcell::fd::ReconstructionMethod::DimByDim);
 
-  if (vars_needed_fixing) {
+  Scalar<DataVector> dg_sqrt_det_spatial_metric{};
+  assign_data(make_not_null(&dg_sqrt_det_spatial_metric), num_dg_pts);
+  evolution::dg::subcell::fd::reconstruct(
+      make_not_null(&get(dg_sqrt_det_spatial_metric)),
+      get(sqrt_det_spatial_metric), dg_mesh, subcell_mesh.extents(),
+      evolution::dg::subcell::fd::ReconstructionMethod::DimByDim);
+
+  if (vars_needed_fixing and
+      (max(get(dg_tilde_d) / get(dg_sqrt_det_spatial_metric)) >
```

Comment:
why check on the DG grid if we are in atmosphere and not the subcell grid? At the very least it should be documented in a C++ comment since naively I'd expect that either subcell or both

Reply:
I'll fix this to check both DG and subcell